pkg/fqdn/name_manager_gc.go

```diff
@@ -44,6 +44,10 @@ var (
 func (n *NameManager) GC(ctx context.Context) error {
 	GCStart := time.Now()
 
+	// Take a snapshot of the *entire* reverse cache, so we can compute the set of
+	// IPs that have been completely removed and safely delete their metadata.
+	maybeStaleIPs := n.cache.dumpReverse()
```

Comment:
I think this is just unclear naming: The `NameManager` stores name->IP and IP->name mappings via `DNSCache`. The latter is named `reverse` internally. The `dumpReverse()` call here is pulling the set of IPs from the `NameManager`. Probably this would be clearer with a name like `n.cache.DumpIPs()` or similar.

Reply:
Agreed, the naming is unclear. All we want is the set of in-use IPs. I'll tighten this up.